File: app/src/main/java/com/ladsoft/bakingapp/fragment/di/RecipesFragmentModule.java
package com.ladsoft.bakingapp.fragment.di;

import com.ladsoft.bakingapp.fragment.RecipesFragment;
import com.ladsoft.bakingapp.mvp.RecipesMvp;
import com.ladsoft.bakingapp.mvp.presenter.RecipesPresenter;

import dagger.Binds;
import dagger.Module;
import dagger.Provides;

@Module
public abstract class RecipesFragmentModule {
    @Binds
    abstract RecipesFragment providesView(RecipesFragment view);

    @Provides
    static RecipesPresenter providesPresenter(RecipesMvp.Model model) {
        return new RecipesPresenter(model);
    }
}

File: app/src/main/java/com/ladsoft/bakingapp/fragment/di/RecipesFragmentModule.kt
package com.ladsoft.bakingapp.fragment.di

import com.ladsoft.bakingapp.fragment.RecipesFragment
import com.ladsoft.bakingapp.mvp.RecipesMvp
import com.ladsoft.bakingapp.mvp.presenter.RecipesPresenter
import dagger.Module
import dagger.Provides

@Module
class RecipesFragmentModule {
    @Provides
    fun providesView(view: RecipesFragment): RecipesFragment = view

    @Provides
    fun providesPresenter(model: RecipesMvp.Model): RecipesPresenter = RecipesPresenter(model)
}